orderCompleted.js

```diff
@@ -0,0 +1,78 @@
+require('dotenv').config();
+const mongoose = require('mongoose');
+const { sendOrderCompletedEmail } = require('./src/mailing/sender');
+const User = require('./src/models/User');
+const { Order } = require('./src/models/Order');
+const CustomError = require('./src/errors');
+const Album = require('./src/models/Album');
+// Connect to the MongoDB database
+async function connectToDatabase() {
+  try {
+    await mongoose.connect(process.env.MONGO_URL, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+    });
+    console.log('Connected to MongoDB');
+  } catch (error) {
+    console.error('Error connecting to MongoDB:', error.message);
+  }
+}
+
+async function sendOrderCompletedEmailToUser(userId) {
+  try {
+    await connectToDatabase();
+    // Find the user by their user ID
+    const user = await User.findById(userId);
+
+    if (!user) {
+      // Throw a NotFoundError if the user is not found
+      throw new CustomError.NotFoundError(`No user was found`);
+    }
+    const username = user.username;
+    console.log('User ID:', user._id);
+    // Check if the user has an order with the 'payment_successful' status
+    const order = await Order.findOne({
+      user: user._id,
+      orderStatus: 'payment_successful',
+    });
+
+    console.log('ORDER:', order._id);
```

Comment:
This can fail if `order` is undefined/null if no matching order was found in line 34. Can either move this line until after the if block lines 40-44 because at that point we _know_ we've found an order. Or we can use optional chaining here `order?._id`